### ISSUE TITLE
Disabling data collection in storage-sync-v2.*

### ISSUE THREAD

**Issue by Gitoffthelawn** [arkenfox/user.js#1187]
I do not have a Firefox account, and I do not sync Firefox with anything.  Yet, much potentially private data is stored in `storage-sync-v2.sqlite`.  And I noticed `storage-sync-v2.sqlite-wal` can grow to be well over 25MB (!) in size.

Would it be reasonable/possible to add prefs to user.js to prevent Firefox from storing data (or at least limiting storage of data) in `storage-sync-v2.*`?

**Comment by Thorin-Oakenpants** [comment]
did you look inside the sqlite? It's used by extensions. Just delete them all so you start clean. Mine was 5mb and had some references to HTTPS-Everywhere and some other extensions I no longer use. It came back (empty) on a restart, because one of my extensions has sync capability (Note Taker)

**Comment by Gitoffthelawn** [comment]
@Thorin-Oakenpants Yep, I looked inside before posting.  As you mentioned, it's all data/settings from extensions, which often includes private information.

IIRC, I deleted them all a year or two ago, and some extensions lost all their data/settings.  I don't remember all the details, but it had something to do with either a FF issue or some extensions being written to store their data in the sync storage.  That's why I was thinking adding a setting to simply disable that storage could workaround the FF issue or perhaps force/encourage extensions to not store their data in those sync databases.

**Comment by Thorin-Oakenpants** [comment]
IDK about losing ext data, I backed up my portable before I wiped mine, and didn't lose anything. I have no idea how ext devs use the API or how it should be used. I would have thought that persistent data is stored elsewhere, the sync is incoming and outgoing, but not my area

What private info are you keeping in extensions .. sounds iffy. Password managers should be encrypted data. Other than that, what are you talking about?
